src/components/InputProblem.tsx

```diff
@@ -105,11 +121,54 @@ export const InputProblem = ({
     return trimmedInput.toLowerCase() === trimmedAnswer.toLowerCase()
   }
 
+  const setPersistedState = async (): Promise<void> => {
+    try {
+      if (contentId === undefined || persistor === undefined) {
+        return
+      }
+
+      if (response !== '' || inputDisabled || retriesAllowed !== 0) {
+        const newPersistedData: PersistorData = { userResponse: response, inputDisabled, retriesAllowed, feedback }
+        await persistor.put(contentId, JSON.stringify(newPersistedData))
+      }
+    } catch (err) {
+      console.error(err)
+    }
+  }
+
+  const getPersistedState = async (): Promise<void> => {
+    try {
+      if (contentId === undefined || persistor === undefined) {
+        setPersistorGetStatus(PersistorGetStatus.Success)
+        return
+      }
+
+      const persistedState = await persistor.get(contentId)
+      if (persistedState !== null) {
+        const parsedPersistedState = JSON.parse(persistedState)
+        setResponse(parsedPersistedState.userResponse)
+        setInputDisabled(parsedPersistedState.inputDisabled)
+        setRetriesAllowed(parsedPersistedState.retriesAllowed)
+        setFeedback(parsedPersistedState.feedback)
+      }
+      setPersistorGetStatus(PersistorGetStatus.Success)
+    } catch (err) {
+      setPersistorGetStatus(PersistorGetStatus.Failure)
+    }
+  }
+
+  useEffect(() => {
+    setPersistedState().catch(() => { })
+    getPersistedState().catch(() => { })
+  }, [response, inputDisabled, retriesAllowed])
+
   const handleSubmit = async (values: InputFormValues, { setFieldError }: FormikHelpers<InputFormValues>): Promise<void> => {
```

Comment:
While I think this submit handler is getting too complicated and we should see if we can't refactor / simplify in some way, from a functional standpoint I think it is very similar to what you had before but maintains the pattern that we "save" on submit:

```javascript
  const handleSubmit = async (values: InputFormValues, { setFieldError }: FormikHelpers<InputFormValues>): Promise<void> => {
    let correct = false
    let finalAttempt = false
    const attempt = retriesAllowed + 1

    const setPersistedState = async (persistorData: PersistorData): Promise<void> => {
      if (contentId === undefined || persistor === undefined) {
        return
      }
      await persistor.put(contentId, JSON.stringify(persistorData))
    }

    if (values.response.trim() === '') {
      if ((comparator.toLowerCase() === 'integer') || (comparator.toLowerCase() === 'float')) {
        setFieldError('response', NUMERIC_INPUT_ERROR)
      } else {
        setFieldError('response', NON_EMPTY_VALUE_ERROR)
      }
      return
    }

    if (evaluateInput(values.response, solution)) {
      correct = true
      try {
        await setPersistedState({ userResponse: values.response, inputDisabled: true, retriesAllowed, feedback: correctResponse })
      } catch (error) {
        setFieldError('response', 'Error saving response. Please try again.')
        return
      }

      setFeedback(correctResponse)
      solvedCallback()
      setInputDisabled(true)
    } else if (retryLimit === 0 || retriesAllowed !== retryLimit) {
      const feedback = determineFeedback(values.response, encourageResponse, answerResponses, evaluateInput)
      try {
        await setPersistedState({ userResponse: values.response, inputDisabled, retriesAllowed: retriesAllowed + 1, feedback })
      } catch (error) {
        setFieldError('response', 'Error saving response. Please try again.')
        return
      }

      setRetriesAllowed(currRetries => currRetries + 1)
      setFeedback(feedback)
      allowedRetryCallback()
    } else {
      try {
        await setPersistedState({ userResponse: values.response, inputDisabled: true, retriesAllowed: retriesAllowed + 1, feedback: attemptsExhaustedResponse })
      } catch (error) {
        setFieldError('response', 'Error saving response. Please try again.')
        return
      }

      setRetriesAllowed(currRetries => currRetries + 1)
      setFeedback(attemptsExhaustedResponse)
      exhaustedCallback()
      setInputDisabled(true)
      finalAttempt = true
    }

    setResponse(values.response)

    if (onProblemAttempt !== undefined) {
      onProblemAttempt(
        values.response,
        correct,
        attempt,
        finalAttempt,
        contentId
      )
    }
  }
```

Admittedly, we don't "save" clearing the feedback, but I think that's fair since it's more of an ephemeral state than the "persisted" state in my mind. Said differently, I think the "persisted" contract should be strictly on an attempt level, and not the "`onChange`" level (both for simplicity and to minimize server load impacts of writes).